kofta/src/vscode-webview/Routes.tsx

```diff
@@ -46,6 +50,30 @@ export const Routes: React.FC<RoutesProps> = () => {
   const [, setFollowingMap] = useAtom(setFollowingMapAtom);
   const [, setFollowingOnline] = useAtom(setFollowingOnlineAtom);
   const [, setInviteList] = useAtom(setInviteListAtom);
+  const [me] = useAtom(meAtom);
+
+  const { open, iAmMentioned } = useRoomChatStore();
+
+  // useEffect, need using me AND iAmMentioned
+  useEffect(() => {
+    addMultipleWsListener({
+      new_chat_msg: ({ msg }) => {
+        useRoomChatStore.getState().addMessage(msg);
+        if (
+          (!open || !document.hasFocus()) &&
+          !!msg.tokens.filter(
+            (t: RoomChatMessageToken) =>
+              t.t === "mention" &&
+              t.v?.toLowerCase() === me?.username?.toLowerCase()
+          ).length
+        ) {
+          useRoomChatStore.getState().setIAmMentioned(iAmMentioned + 1);
+        }
+      },
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [me, iAmMentioned, open]);
```

Comment:
instead of creating a new useEffect, do `const {open, iAmMentioned} = useRoomChatStore.getState()` to get the most recent values

Reply:
Great thanks.

Reply:
@benawad I have to `useEffect` for the `me` atom.

Reply:
Should I remove this logic from Routes.tsx and put it somewhere, that way we can get the latest value of `me` I think.

Reply:
I see, lets do a quick fix by sticking current me value in a ref
```
const meRef= useRef(me);
meRef.current = me
```

Reply:
@benawad Okay it works that way 🎉